Documentation/rfc/v3api.md

```diff
@@ -0,0 +1,459 @@
+## Design
+
+1. Flatten binary key-value space
+    
+2. Keep the event history until compaction
+    - access to old version of keys
+    - user controlled history compaction
+    
+3. Support range query
+    - Pagination support with limit argument
+    
+4. Replace TTL key with Lease
+    - more efficient/ low cost keep alive
+    - a logical group of TTL keys
+    
+5. Replace CAS/CAD with multi-object Tnx
+    - MUCH MORE powerful and flexible
```

Comment:
I'm curious about this - what will the effective difference be with this change? Will this change allow bulk loading (`PUT`'s)? If so, what will the data format need to be?